### ISSUE TITLE
Boolean button behavior with confirmation

### ISSUE THREAD

**Issue by georgweiss** [ControlSystemStudio/phoebus#2423]
If the Boolean Button widget is configured as "push" and to show a confirmation dialog (on set), the button does not return to default value after positive confirmation. As this is different compared to running without confirmation I assume this is a bug, but want to confirm before I tackle it.

**Comment by georgweiss** [comment]
Thanks, but not sure this helps me.
What is the intension using the boolean button in "push" mode? From the example OPIs it seems that 1 is written to the specified bit, then immediately afterwards 0 is written.

**Comment by kasemir** [comment]
"Push" writes 1 on mouse-down, then 0 on mouse-release. The intended usage is for example moving a motor as long as the button is pushed down, then stop when released. So typically held down for a little while.

"Toggle" changes between 0 and 1 on each click. For example used to turn a power supply on/off.

**Comment by georgweiss** [comment]
OK thanks.
So I guess using push together with confirmation dialog in practice turns the push mode to a toggle mode.

**Comment by kasemir** [comment]
It can't really work with "push". You push down, and should keep it pressed down for a while. But you can't, because you need to handle the confirmation dialog...

**Comment by georgweiss** [comment]
So maybe we should disable use of confirmation dialog with push?

**Comment by kasemir** [comment]
Ideally, yes, but it's a little tricky. The editor fundamentally just lists widgets and their properties. It doesn't "understand" what they do. You can configure both the background and the foreground color of a text to be black, the editor won't explain to you why you no longer see anything. Likewise, the editor doesn't know about the Bool Button with its toggle mode vs. confirmation behavior. So this would have to be in the bool button model or maybe even representation: If it notices "push" with "confirm", it changes one or both of them to remain compatible, and the editor just shows the current settings.

**Comment by georgweiss** [comment]
Well, if we can agree that "push" shall not support confirmation dialog then I can give it a try...

**Comment by kasemir** [comment]
> "push" shall not support confirmation dialog 

Yes, I'd say there's no practical way to keep pushing down for a while and handle confirmation at the same time.

**Comment by georgweiss** [comment]
So I looked into this and here is a proposal:

If user selects Push/Push inverted, the confirmation dialog setting will automatically set to "no". If user then tries to select confirmation dialog, the value stays unchanged. For this to work I need to change in EnumWidgetPropertyBinding#L42:

```
if (updating ||  jfx_node.isFocused())
``` 
to
```
if (updating)
``` 

Question is also whether to launch an alert with suitable message if user wants to configure an unsupported combination of mode and confirmation dialog.

**Comment by georgweiss** [comment]
Ideally I'd like to disable the confirmation dialog drop-down if user selects anything but toggle. Is there an API for that?
But even so, supporting existing display files using the unsupported combination complicates matters.

I was planning to put everything in the ```BoolButtonRepresentation```, yes. Maybe better to simply launch a dialog when user selects an unsupported combination, and then revert back to previous value for either mode or confirmation dialog property.

**Comment by georgweiss** [comment]
Since a display could contain multiple such boolean buttons, it would be quite annoying to have to dismiss all dialogs. Moreover, this issue was discovered in a display from a user that is pretty new to CS Studio, so let's assume it's a beginner's mistake.

In short: existing displays would show dialog only of these properties are changed.
